internal/kldevents/submanager.go

```diff
@@ -125,6 +126,20 @@ func (s *subscriptionMGR) Subscriptions(ctx context.Context) []*SubscriptionInfo
 	return l
 }
 
+func (s *subscriptionMGR) setInitialBlock(i *SubscriptionInfo, initialBlock string) error {
+	// Check initial block number to subscribe from
+	if initialBlock == "" || initialBlock == FromBlockLatest {
+		i.FromBlock = FromBlockLatest
+	} else {
+		var bi big.Int
+		if _, ok := bi.SetString(initialBlock, 0); !ok {
```

Comment:
Why are we using base `0` to test validity and base `10` to set `FromBlock`?

Reply:
The first part *sets* the value from the input string, with a friendly way of base `0` - meaning `0x1234` would work as hex, and `1234` would work as decimal.

However, the `FromBlock` is then consistently stored as a base-10 string. So the serialized format is _always_ base 10, regardless of what the user submitted